docs/code-style-guide.md

```diff
@@ -62,6 +62,32 @@ The vast majority of the things noted in this document will apply to `C` code as
 
 * Functions that take no arguments must be declared as `foo(void)` **not** `foo()`, as the second option allows `foo` to take anything as it's arguments ([reference](https://softwareengineering.stackexchange.com/questions/286490/what-is-the-difference-between-function-and-functionvoid/286494))
 
+* Functions that return values via argument must have all parameters labelled as `[in]` or `[out]` in the javadoc, with all `[out]` coming after `[in]` if possible
```

Comment:
What about arguments that are modified in place? Should they be labelled `[in/out]`? (I think that's what we have currently)

Reply:
Good point. Added.